### PR TITLE
Master release 4.3.00

### PR DESCRIPTION
**Pull request by ndellingwood**

### CODE REVIEW

**Comment by ndellingwood** [comment]
Trilinos snapshot PR: https://github.com/trilinos/Trilinos/pull/12879
Do not merge until confirmation that PR has passed testing and is approved

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,99 @@
...
+- Syr2 [\#1942](https://github.com/kokkos/kokkos-kernels/pull/1942)
+
+#### LAPACK updates
+- Adding cuSOLVER [\#2038](https://github.com/kokkos/kokkos-kernels/pull/2038)
+  - Fix for MAGMA with CUDA [\#2044](https://github.com/kokkos/kokkos-kernels/pull/2044)
+- Adding rocSOLVER [\#2034](https://github.com/kokkos/kokkos-kernels/pull/2034)
+  - Fix rocSOLVER issue with Trilinos dependency [\#2037](https://github.com/kokkos/kokkos-kernels/pull/2037)
+- Lapack - SVD [\#2092](https://github.com/kokkos/kokkos-kernels/pull/2092)
+  - Adding benchmark for SVD [\#2103](https://github.com/kokkos/kokkos-kernels/pull/2103)
+  - Quick return to fix cuSPARSE and improve performance [\#2107](https://github.com/kokkos/kokkos-kernels/pull/2107)
```
Comment (brian-kelley) [inline CHANGELOG.md:18]:
```suggestion
  - Quick return to fix cuSOLVER and improve performance [\#2107](https://github.com/kokkos/kokkos-kernels/pull/2107)
```

**Review by brian-kelley — APPROVED** [review]

**Comment by ndellingwood** [comment]
I'm going to force push some changes, the merge resulted in an artifact that showed up with the Trilinos

**Comment by ndellingwood** [comment]
DOCS GA failure due to prohibited testing of release (4.3.00) against kokkos' develop branch (4.3.99), which is outside range of versions supported by the 4.3.0 release

**Review by lucbv — APPROVED** [review]

**Comment by ndellingwood** [comment]
changelog update PR to develop is up: https://github.com/kokkos/kokkos-kernels/pull/2169

**Comment by ndellingwood** [comment]
Trilinos snapshot PR https://github.com/trilinos/Trilinos/pull/12879 passed CI

**Comment by ndellingwood** [comment]
Fixup https://github.com/kokkos/kokkos-kernels/pull/2171 cherry-picked, and snapshot of this updated branch made to Trilinos https://github.com/trilinos/Trilinos/pull/12892 with the fix

**Comment by ndellingwood** [comment]
This should be ready for merge and tagging :)

**Pull request merged**